starter-code/basic-algorithms.js

```diff
@@ -1,7 +1,94 @@
+  console.log(name2[i]);
+}
+
+// lexicographic order alfabede önce geliyorsa 1 sonra geliyorsa -1 aynı yerdelerse 0.!!  örnk a localcompare b--> -1 çünkü a bden önce
+var lexicographic = console.log("hacker1".localeCompare("name2"));
+
+if (lexicographic < 0) {
+  console.log("The driver´s name goes first");
+} else if (lexicographic > 0) {
+  console.log("The navigator goes first definitely");
+} else {
+  console.log("What?! You both got the same name?");
+}
+
+// Palindrome
+
+var palindrome = prompt("Enter a word and I check it if is Palindrome!");
+var array = [];
+array = palindrome.split("");
+
+array.reverse();
+
+var palindrome2 = "";
+
+palindrome2 = array.join("");
+
+console.log(palindrome);
+console.log(palindrome2);
+
+if (palindrome === palindrome2) {
+  alert("It is PALINDROME");
+} else alert("Nope it is not a PALINDROME");
+
+// Paragraph thingy : count Number of Letters
+var paragraph =
+  "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.";
+
+console.log(paragraph.length);
+
+// paragraph thingy count of words  Not: \W matches anything that is not a latin letter or number or an underscore”
+var paragraph =
+  "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.";
+console.log(paragraph.split(/\W+/).length);
+
+// count et word
+
+var par =
+  "Lorem  et Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.";
+var words = par.split("");
+var count = 0;
+if (words === "et") count++;
```

Comment:
ici l'idée est bonne mais tu incrémentes (ajouter une valeur a chaque fois) count avec... 0
donc il va rester a 0.
var words = par.split(" "); (<= NE PAS OUBLIER L'ESPACE entre les " " car cela te permet de split entre chaque mot et non entre chaque lettre. Comme on cherche le nombre de "et" !)
var newArray = [];
for(var count = 0; count < words.length; count ++) {
 if (words[count] === "et") {
  newArray.push(count)
  }
console.log(newArray.length)
}
On crée un array vide et on fait un for loop qui va durer aussi longtemps qu'il y a de mot dans words.
ensuite si un mot dans words correspond a "et" on le push dans notre array vide avec le method .push.
et on console.log la longueur de notre nouvelle array qui comprend tous les "et" !